src/vsr/client.zig

```diff
@@ -414,10 +417,10 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
 
             // Newly made Requests are populated and potentially pushed through VSR if there exists
             // no currently inflight request.
-            // A Batch is newly made if its demux node is the same one inlined in the Request.
-            // NOTE: This sets the batch.request.message.request number.
-            if (batch.demux == &batch.request.demux) {
-                assert(batch.request.demux_queue.empty());
+            // A Batch is newly made if its Request only contains its Demux.
+            // NOTE: This sets the `batch.request.message.request` number.
```

Comment:
```zig
// NOTE: This sets the `batch.request.message.request` number.
```
I don't think this comment is up to date, the request number was already set during `batch_get`.

Also, we can swap the order of `self.register()` with the `header` initialization, so we don't need to use `.request = undefined` when initializing the header anymore.